src/xpath.h

```diff
@@ -223,6 +223,13 @@ struct lyxp_set {
         struct lyxp_set_scnode {
             struct lysc_node *scnode;
             enum lyxp_node_type type;
+
+#define LYXP_SET_SCNODE_TRAVERSED_IN            -2
+#define LYXP_SET_SCNODE_TRAVERSED_NOSTART_NOTIN -1
+#define LYXP_SET_SCNODE_TRAVERSED_NOTIN          0
+#define LYXP_SET_SCNODE_IN                       1
+#define LYXP_SET_SCNODE_IN_JUST_ADDED            2
+#define LYXP_SET_SCNODE_NOTIN                    3
```

Comment:
I am not sure about these, I have nothing against using named constants instaed of numbers so that the code is self-explaining but in this case it is not that simple. Value `3` and any higher ones are all used for predicates, there is not just one. So, it may be better to just leave it as it is.

Reply:
I've tried to improve the description here. But I don't see any problem with using `LYXP_SET_SCNODE_NOTIN` as you describe and as previously the value `3` was used. Any higher value was assigned by incrementing the current (or some other) value. What I did in the code was just replacing numbers with the named constants.

Reply:
Okay, then I wanted to say that not all the uses of these values were refactored, now they seem to be (I have not searched for them myself assuming you did it). Then I will only suggest better names for the values.
`LYXP_SET_SCNODE_TRAVERSED_IN` - `LYXP_SET_SCNODE_START`
`LYXP_SET_SCNODE_TRAVERSED_NOSTART_NOTIN` - `LYXP_SET_SCNODE_START_USED`
`LYXP_SET_SCNODE_TRAVERSED_NOTIN` - `LYXP_SET_SCNODE_ATOM`
`LYXP_SET_SCNODE_IN` - `LYXP_SET_SCNODE_ATOM_CTX`
`LYXP_SET_SCNODE_IN_JUST_ADDED` - `LYXP_SET_SCNODE_ATOM_NEW_CTX`
`LYXP_SET_SCNODE_NOTIN` - `LYXP_SET_SCNODE_ATOM_PRED_CTX`
They are not perfect but with the commen it should be fairly clear.

Reply:
yes, first I forgot to go through all uses of the `in_ctx` member, but it should be fixed since the previous update.

ok, I'm going to change the names.

Reply:
fixed